packages/amplify_datastore/lib/amplify_datastore.dart

```diff
@@ -53,6 +57,114 @@ class AmplifyDataStore extends DataStorePluginInterface {
     return streamWrapper.datastoreStreamController;
   }
 
+  @override
+  Future<void> configure({
+    AmplifyConfig? config,
+    required AmplifyAuthProviderRepository authProviderRepo,
+  }) async {
+    if (config == null) {
+      throw ConfigurationError('No Amplify config provided');
+    }
+
+    final authPlugin = Amplify.Auth.plugins.firstOrNull;
+    if (authPlugin != null) {
+      // Configure this plugin to act as a native iOS/Android plugin.
+      final nativePlugin = _NativeAmplifyAuthCognito();
+      NativeAuthPlugin.setup(nativePlugin);
+      final nativeBridge = NativeAuthBridge();
+      try {
+        await nativeBridge.addAuthPlugin();
+      } on PlatformException catch (e) {
+        if (e.code.contains('AmplifyAlreadyConfiguredException') ||
+            e.code.contains('AlreadyConfiguredException')) {
+          throw const AmplifyAlreadyConfiguredException(
+            AmplifyExceptionMessages.alreadyConfiguredDefaultMessage,
+            recoverySuggestion:
+                AmplifyExceptionMessages.alreadyConfiguredDefaultSuggestion,
+          );
+        }
+        throw ConfigurationError(
+          e.message ?? 'An unknown error occurred',
+          underlyingException: e,
+        );
+      }
+      // Update the native cache for the current user on hub events.
+      Future<void> updateCurrentUser(AuthUser? currentUser) async {
+        NativeAuthUser? nativeUser;
+        if (currentUser != null) {
+          nativeUser = NativeAuthUser(
+            userId: currentUser.userId,
+            username: currentUser.username,
+          );
+        }
+        await nativeBridge.updateCurrentUser(nativeUser);
+      }
+
+      try {
+        final currentUser = await Amplify.Auth.getCurrentUser();
+        await updateCurrentUser(currentUser);
+      } on Exception {
+        await updateCurrentUser(null);
+      }
+      Amplify.Hub.listen(HubChannel.Auth, (AuthHubEvent event) {
+        updateCurrentUser(event.payload);
+      });
+    }
+
+    final apiPlugin = Amplify.API.plugins.firstOrNull;
+    if (apiPlugin != null) {
```

Comment:
In addition to the API plugin presence check, we can also check if `amplifyconfiguration.dart` has a valid `GraphQL` typed API.

Although the native API plugin surely can handle missing config gracefully, I think it's better to not let the native layer throw an unmanaged exception. (especially that iOS doesn't print the exception info in Flutter debug console.)

Reply:
The API category checks that there is something present in the config for API. Where you thinking of something other than that?

https://github.com/aws-amplify/amplify-flutter/blob/81b8728b66c69f3be28f50521d4d07ece055783c/packages/api/amplify_api_dart/lib/src/api_plugin_impl.dart#L74-L85

Reply:
Yes I meant to add more specific check for DataStore checking if a GraphQL API exists in the config while automatically adding/configuring the native API plug-in.